lib/collection/src/common/file_utils.rs

```diff
@@ -57,3 +57,33 @@ pub async fn move_file(from: impl AsRef<Path>, to: impl AsRef<Path>) -> Collecti
     }
     Ok(())
 }
+
+/// Guard, that ensures that file will be deleted on drop.
+pub struct FileCleaner {
+    path: Option<PathBuf>,
+}
+
+impl FileCleaner {
+    pub fn new(path: impl Into<PathBuf>) -> Self {
+        Self {
+            path: Some(path.into()),
+        }
+    }
+
+    pub fn delete(&mut self) {
+        if let Some(path) = &self.path {
+            // Ignore errors, because file can be already deleted.
+            let _ = std::fs::remove_file(path);
```

Comment:
Quick question that just popped in my head before merging:

This fails on directories, and yes I know this is a **file** cleaner. But should we make this more robust anyway and do `path.is_dir()` with `std::fs::remove_dir_all(path)`?

```suggestion
            // Ignore errors, because file can be already deleted.
            if path.is_dir() {
                let _ = std::fs::remove_dir_all(path);
            } else {
                let _ = std::fs::remove_file(path);
            }
```